R/mainTests.r

```diff
@@ -2539,3 +2543,194 @@ indicator_dp_completed <- function(meta) {
+    )
+  } else if (length(ethnicity_columns) == 1) {
+    output <- list(
+      "message" = paste0(
+        paste(ethnicity_columns, collapse = "', '"), " appears to relate to ethnicity data, but does not conform to the standard col_name conventions: ",
+        paste(ethnicity_standard_headers, collapse = ", "),
+        "."
+      ),
+      "result" = "FAIL"
+    )
+  } else {
+    output <- list(
+      "message" = paste0(
+        "The following columns appear to relate to ethnicity data, but do not conform to the standard col_name conventions: <br> - '",
+        paste(ethnicity_columns, collapse = "', '"), "'. <br> - These should take the form of one of the following: ",
+        paste(ethnicity_standard_headers, collapse = ", "),
+        "."
+      ),
+      "result" = "FAIL"
+    )
+  }
+  return(output)
+}
+
+ethnicity_values <- function(data) {
+  # First find any ethnicity type columns that don't have the standard col_names
+  if ("ethnicity_major" %in% colnames(data) & "ethnicity_minor" %in% colnames(data)) {
+    ethnicity_nonstandard <- data %>%
+      mutate(ethnicity_combined = paste(ethnicity_major, ethnicity_minor, sep = ", ")) %>%
+      select(ethnicity_combined) %>%
+      unique() %>%
+      filter(!grepl(
+        paste(paste(ethnicity_standard_values$ethnicity_major, ethnicity_standard_values$ethnicity_minor, sep = ", "), collapse = "|"),
+        ethnicity_combined
+      )) %>%
+      pull(ethnicity_combined)
+    value_type <- "combination"
+  } else if ("ethnicity_major" %in% colnames(data)) {
+    ethnicity_nonstandard <- data %>%
+      select(ethnicity_major) %>%
+      unique() %>%
+      filter(!grepl(
+        paste(ethnicity_standard_values$ethnicity_major, collapse = "|"),
+        ethnicity_major
+      )) %>%
+      pull(ethnicity_major)
+    value_type <- "value"
+  } else if ("ethnicity_minor" %in% colnames(data)) {
+    ethnicity_nonstandard <- data %>%
+      select(ethnicity_minor) %>%
+      unique() %>%
+      filter(!grepl(
+        paste(ethnicity_standard_values$ethnicity_minor, collapse = "|"),
+        ethnicity_minor
+      )) %>%
+      pull(ethnicity_minor)
+    value_type <- "value"
+  } else {
+    ethnicity_nonstandard <- c()
+  }
+  if (length(ethnicity_nonstandard) == 0) {
+    output <- list(
+      "message" = "No ethnicity entry issues found.",
+      "result" = "PASS"
+    )
+  } else if (length(ethnicity_nonstandard) == 1) {
+    output <- list(
+      "message" = paste0(
+        "The ethnicity filter ", value_type, " '",
+        paste(ethnicity_nonstandard, collapse = "', '"),
+        "' does not conform to the GSS standards. Please cross check against the <a href='https://rsconnect/rsc/stats-production-guidance/ud.html#Ethnicity'>published standards</a>."
+      ),
+      "result" = "ADVISORY"
+    )
+  } else {
+    output <- list(
+      "message" = paste0(
+        "The following ethnicity filter ", value_type, "s do not conform to the GSS standards: <br> - '",
```

Comment:
I think this should be filter**s** before the value_type (typo, s is currently after?)

Reply:
The variable value_type can be either "combination" or "value", so the above should get rendered as "...filter values do not..." or "...filter combinations do not..."